repo/blob/throttling/throttling_reader.go

```diff
@@ -0,0 +1,76 @@
+package throttling
+
+import (
+	"context"
+	"io"
+
+	"github.com/kopia/kopia/repo/blob"
+)
+
+// ThrottledBytes acts as blob.Bytes but ensures that reads are throttled.
+type ThrottledBytes struct {
+	data      blob.Bytes
+	throttler Throttler
+	ctx       context.Context //nolint:containedctx
+}
+
+var _ blob.Bytes = &ThrottledBytes{} // check we meet the interface we're trying to meet
+
+// Length returns the length.
+func (t ThrottledBytes) Length() int {
+	return t.data.Length()
+}
+
+// WriteTo writes to the writer while meeting the rate limit.
+func (t ThrottledBytes) WriteTo(w io.Writer) (int64, error) {
+	// Use Copy & t.Reader rather than delegate to t.data.WriteTo to ensure we rate limit
+	return io.Copy(w, t.Reader()) //nolint:wrapcheck
```

Comment:
`io.Copy()` will allocate unnecessary memory for temporary buffers. We have `iobuf.Copy()` and `iobuf.JustCopy()` which avoids this.